mmv1/products/networksecurity/ServerTlsPolicy.yaml

```diff
@@ -0,0 +1,174 @@
+      update_minutes: 30
+      delete_minutes: 30
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+autogen_async: true
+import_format: ['projects/{{project}}/locations/{{location}}/serverTlsPolicies/{{name}}']
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: 'network_security_server_tls_policy_basic'
+    primary_resource_id: 'default'
+    vars:
+      resource_name: 'my-server-tls-policy'
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: 'network_security_server_tls_policy_advanced'
+    primary_resource_id: 'default'
+    vars:
+      resource_name: 'my-server-tls-policy'
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    required: true
+    url_param_only: true
+    immutable: true
+    description: |
+      Name of the ServerTlsPolicy resource.
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    default_value: 'global'
+    description: |
+      The location of the server tls policy.
+      The default value is `global`.
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      Time the ServerTlsPolicy was created in UTC.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      Time the ServerTlsPolicy was updated in UTC.
+    output: true
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: 'labels'
+    description: Set of label tags associated with the ServerTlsPolicy resource.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      A free-text description of the resource. Max length 1024 characters.
+  - !ruby/object:Api::Type::Boolean
+    name: 'allowOpen'
+    description: |
+      This field applies only for Traffic Director policies. It is must be set to false for external HTTPS load balancer policies.
+      Determines if server allows plaintext connections. If set to true, server allows plain text connections. By default, it is set to false. This setting is not exclusive of other encryption modes. For example, if allowOpen and mtlsPolicy are set, server allows both plain text and mTLS connections. See documentation of other encryption modes to confirm compatibility.
+      Consider using it if you wish to upgrade in place your deployment to TLS while having mixed TLS and non-TLS traffic reaching port :80.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'serverCertificate'
+    description: |
+      Defines a mechanism to provision client identity (public and private keys) for peer to peer authentication. The presence of this dictates mTLS.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'grpcEndpoint'
```

Comment:
I saw in the docs that `grpcEndpoint` and `certificateProviderInstance` are part of a union field and are mutually exclusive. What do you think about using [the exactly_one_of attribute](https://googlecloudplatform.github.io/magic-modules/docs/how-to/add-mmv1-resource/#exactly-one-of) to stop users trying to set both fields? Another approach is to include a note like 'x cannot be set at the same time as y' in the field descriptions, because users can attempt to use a Terraform resource without reading the relevant Google API docs first.

Reply:
Yes thanks for the suggestion! Added this check